pubtools/_quay/command_executor.py

```diff
@@ -44,7 +44,9 @@ def skopeo_login(self, username=None, password=None):
             )
         LOG.info("Logging in to Quay with provided credentials")
 
-        cmd_login = "skopeo login -u {0} --password-stdin quay.io".format(username)
+        cmd_login = (
+            "skopeo login --authfile $HOME/.docker/config.json -u {0} --password-stdin quay.io"
+        ).format(username)
```

Comment:
Look like this could allow a command injection using a constructed username.

Reply:
Not sure I understand, could you elaborate?

Reply:
What happens if someone "accidentally" passes a username with a space?

For the command injection: What about setting username to `;rm -rf xyz;` or something?

Not sure if it's possible to do the same by overriding the `HOME` env variable (used in the command).

Reply:
IMO this is up to the remote host to guard against. If the user is able to supply a username/password or a private key to access the remote host, then they can simply do rm -rf directly with no injection required.